### ISSUE TITLE
Poisson regression

### ISSUE THREAD

**Issue by MaxHalford** [creme-ml/creme#124]
Poisson regression is great for predicting counts, and I think it would nice to have to it in `creme`. In principle this should reuse most of the code from `LinearRegression`. A good way to validate the model would be to show that it performs better on the `fetch_bikes` dataset.

**Comment by jklaise** [comment]
Given the code reuse, I wonder if there should be a unified API for generalized linear models?

**Comment by MaxHalford** [comment]
Indeed I was thinking the same thing. I think that a good idea would be to only have a `LinearRegression` and a `LogisticRegression`. Adding more classes adds cognitive load to users, which I don't believe should be our philosophy. On the other hand, if we had a single `LinearModel` class then it would be full of ifs to handle the regression/classification differences.

**Comment by jklaise** [comment]
I think it's a good idea to keep the two most widely used models as separate classes. For other we could do something like `GLMRegressor` and `GLMClassifier`, of course these should also implement linear and logistic regression a as base cases respectively.

**Comment by MaxHalford** [comment]
I took a go at making a `GLM` class, but in the end I don't think it's a good idea. The trick is that we want to be able to treat regression differently from classification because of the intercept. This is specific to `creme` and the way we do things, and is a little complicated to explain.

As for Poisson regression, we simply want to implement the Poisson loss function in the `optim` module.

**Comment by jklaise** [comment]
@MaxHalford interesting, is there some reading material related to this?

**Comment by MaxHalford** [comment]
This [page](https://scikit-learn.org/stable/modules/sgd.html#sgd) is good, especially the last paragraph. The way Léon Bottou (and scikit-learn) propose to update the intercept works well for classification. But in practice I've observed that for regression using a running average of the target works better.

**Comment by MaxHalford** [comment]
Done! I used [VW's example notebook](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/python/examples/poisson_regression.ipynb) for testing.
